Shared/Stores/PeopleStore.swift

```diff
@@ -0,0 +1,174 @@
+//
+//  UserStore.swift
+//  Jibber
+//
+//  Created by Benji Dodgson on 10/22/21.
+//  Copyright © 2021 Benjamin Dodgson. All rights reserved.
+//
+
+import Foundation
+import Combine
+import ParseLiveQuery
+import Parse
+import StreamChat
+import Contacts
+
+/// A store that contains all people that the user has some relationship with. This could take the form of a directly connected Jibber chat user
+/// or it could just be another person that has been invited but not yet joined Jibber.
+class PeopleStore {
+
+    static let shared = PeopleStore()
+    private var cancellables = Set<AnyCancellable>()
+
+    @Published var personUpdated: PersonType?
+    @Published var userDeleted: User?
+
+    var people: [PersonType] {
+        var allPeople: [PersonType] = self.users
+        let contactPeople: [PersonType] = self.contacts.map { contact in
+            return Person(withContact: contact)
+        }
+        allPeople.append(contentsOf: contactPeople)
+        return allPeople
+    }
+    private(set) var users: [User] = []
+    private(set) var contacts: [CNContact] = []
+
+    private var initializeTask: Task<Void, Never>?
+
+    func initializeIfNeeded() async {
+        // If we already have an initialization task, wait for it to finish.
+        if let initializeTask = self.initializeTask {
+            await initializeTask.value
+            return
+        }
+
+        // Otherwise start a new initialization task and wait for it to finish.
+        self.initializeTask = Task {
```

Comment:
should this be weak?

Reply:
@Benjamindodgson It wouldn't hurt, but because this is a static singleton it'll never be deallocated.